src/io/dataset.cpp

```diff
@@ -658,7 +658,9 @@ void Dataset::ConstructHistograms(const std::vector<int8_t>& is_feature_used,
         // feature is not used
         auto data_ptr = hist_data + group_bin_boundaries_[group];
         const int num_bin = feature_groups_[group]->num_total_bin_;
-        std::memset(data_ptr + 1, 0, (num_bin - 1) * sizeof(HistogramBinEntry));
+        for (int i = 0; i < num_bin; ++i) {
+            data_ptr[1 + i] = {};
```

Comment:
I think the previous code means Initializing objects to all zeroes, so I initialize each objects with `{}`.
I'm not sure this is right.

Reply:
cf. https://stackoverflow.com/questions/2837854/initializing-an-object-to-all-zeroes

Reply:
I think the memset is faster than for loop. 
can you try to change line 659 to a "void*" pointer ?

Reply:
yes, I think so.
I'll try "void*" pointer.

Reply:
okay, then you can `+1` when changing its type.